### DISCUSSION TITLE
Headings, Pages & Navigation Tree

### DISCUSSION THREAD

**Discussion by michaelkowalczyk** [LiaScript/LiaScript#175] [General]
Thank you for providing LiaScript, we are considering using it in higher education to create OERs including H5P elements.

For the title of the OER we use H1, the rest of the chapters are styled using H2 and H3 headings.

H1 Title
H2 Chapter 1
H3 Subchapter 1.1

Current observation:
1. Adding any heading (H1, H2, H3 ...) will result in a new "Page" and "(Sub)Chapter" in the left navigation panel.
2. The navigation tree (sub)chapters are not collapsible. Using many headings can result in a rather long list which can be overwhelming to navigate.
3. Changing the "Presentation Mode" between "Textbook", "Presentation" or "Slides" does not change the appearance.

Related to the above, would it be possible to...
- Allow sub-headings to be displayed on one page rather than splitting each heading into a separate page? 
This could be achieved by giving each "presentation mode" a specific style, e.g. "Textbook" = display H2 and all its sub-chapters on one page.
- Allow "hiding" headings H3 and lower in the navigation tree.

Kind regards 
Michael

**Comment by andre-dietrich** [comment]
Hi, that sounds great ... actually, you can organize your course as you will. The easiest way is to use HTML-tags. Within the first preprocessing, LiaScript will ignore all HTML-content and add this to your slide, within one `<section>` you can provide one header to some content, while `<article>` might be more complex with multiple headers:

``` markdown
# Headers

## Sub Header 1

<section>

### Sub Sub Header 1 (not part of the Table-of-Contents)

todo ...

</section>

<section>

### Sub Sub Header 2 (not part of the Table-of-Contents)

todo ...

</section>

## Sub Header 2

as an alternative you can use

Sub Sub Header (not part of the Table-of-Contents)
=============

and

Sub Sub Sub Header (not part of the Table-of-Contents)
------------------

But this does not follow the original Markdown standard
```

Here is a little demo that you could try out:

[Example](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA32RwU4DMQxE7/mKET13EZyBA6ce6In+gHfjtlGzcZU4rfbvcdqKskJgKYdEb8b2ZIEVk+dcnNvsOTPIzlijhmNknGkqkC3I+5B2KLVf7q84VEB4KDF4fuiaFj0XbQqEAhq0UoxTw3js2UP3PCIku68264+l0q4ghgOj8KBBEiRbbw1D5GZnHs1qkoqBEo5ZTtYJo7TxmFIbp59Qy2UuHimZlOLFvHNuscBn7W+74cm5rrPXl1uvtwZciTml4gVX9PHO/i97/ks2n8EwslwSKCrnRBpO/L1eLezc3Ne9zsrUyd+ZH9zyVzn3XtUCtwy9cEESxVZilHP7BQs67EKyrNaUD17OCUXNnLL/AtGHB2MNAgAA)

I tried to address the usage of semantic correct HTML here:

https://liascript.github.io/course/?https://raw.githubusercontent.com/liaScript/docs/master/README.md#Semantic-Correct-HTML

but there is more to it, actually this could be also handled with a simple `<div>`, but we can add more meaning to the content...

https://www.w3schools.com/html/html5_semantic_elements.asp

I hope this solves your problem ;-)

Kind regards

André

**Reply by michaelkowalczyk** [reply]
Thanks for the pointer howto use Headings inside sections André! 
I should have looked at the Readme in the first place.
Would you consider adding the possibility to make the Sub-Navigation items collapsable in a future version?

**Reply by andre-dietrich** [reply]
This is a long term goal, we did not add it so far, because it interferes with the search... which we want to improve also by enabling previews and fuzzy search or regular expression :see_no_evil: ...
